src/reportengine/api.py

```diff
@@ -0,0 +1,45 @@
+"""
+API for resource engine
+"""
+
+import logging
+import importlib
+
+from reportengine.resourcebuilder import ResourceBuilder
+from reportengine.resourcebuilder import FuzzyTarget
+from reportengine.configparser import Config
+from reportengine.environment import Environment
+
+log = logging.getLogger(__name__)
+
+class API:
+    """The API class"""
+    config_class = Config
```

Comment:
I think it was a mistake in `App` to make these class variables. It means that you have to subclass essentially always. We should make them instance variables instead.